### ISSUE TITLE
yomitan: Could not connect to native MeCab component

### ISSUE THREAD

**Issue by phamthaison1207** [themoeway/yomitan#761]
currently I am trying to switch from the internal yomitan parser to mecab (more accurate) but am stuck here: 
Does anyone know what is going wrong?
tried doing everything that starxeras recommended and it still doesn't work for some reason.
my yomitan version is the latest version 24.12
I installed mecab, ran the mecab installer for yomitan, have this but idk why it's still not working:
![image](https://github.com/themoeway/yomitan/assets/63585568/dcf82773-f205-47f6-8a8c-e07f86c01afc)
![image](https://github.com/themoeway/yomitan/assets/63585568/e544e82b-a4de-4261-985e-ea7bbff8951e)
![image](https://github.com/themoeway/yomitan/assets/63585568/eb77a4cf-9919-4dfc-b52c-d1c897e47960)

**Comment by starxeras** [comment]
I will put this here for future reference. I do not really know how the code in the mecab installer works, as I just changed a few lines to make it (mostly) work with the newest extension. I was told to merge it with TMW for security purposes. Anyway, I am definitely not the most qualified person to help people with it. 

However, I have done some testing with it, so I'll give some tips:
1. Try to reinstall it with your browser completely closed. 
2. If you're using Chrome, go to the Settings page in Yomitan and copy the url that looks almost exactly like this: 
- `chrome-extension://likgccmbimhjbgkjambclfkhldnlhbnn/`
- (Make sure to include the chrome-extension and the  / at the end)
- Then, when reinstalling the mecab installer, paste that exact extension ID when it says "Add more extension IDs" on the console. Press enter and go about as usual. 
- (Do note that if your extension ID is exactly the same as the one pasted above, you shouldn't need to do this)
- (Also, for Firefox, you should be able to do the same thing. Firefox puts the extension ID in a different place, though. I think it's there when you go to your Extensions page and click on Yomitan. It is in the same place where you can check if Yomitan is visible in Incognito windows. It'll look like this: `{XXXXXXXXXXXXXXXX}`. Include the brackets.)
3. Just try to reinstall it multiple times. There might've been a fluke. 

If none of these works, then I sadly don't really know what the issue is. I can definitely try to understand the code, but that'll take time, and I'm pretty busy right now. If there are any more qualified people out there, they will be in #tmw-dev in the discord (or #yomitan-discussions).

**Comment by avc1657** [comment]
> currently I am trying to switch from the internal yomitan parser to mecab (more accurate) but am stuck here: Does anyone know what is going wrong? tried doing everything that starxeras recommended and it still doesn't work for some reason. my yomitan version is the latest version 24.12 I installed mecab, ran the mecab installer for yomitan, have this but idk why it's still not working: ![image](https://private-user-images.githubusercontent.com/63585568/309536293-dcf82773-f205-47f6-8a8c-e07f86c01afc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3NDg1OTgsIm5iZiI6MTcwOTc0ODI5OCwicGF0aCI6Ii82MzU4NTU2OC8zMDk1MzYyOTMtZGNmODI3NzMtZjIwNS00N2Y2LThhOGMtZTA3Zjg2YzAxYWZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA2VDE4MDQ1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3YzMxMDU5MmU3Yzk2OThiMjhjYWJiNzI4MDA2ZTdkYzQwNmU5N2ZmMmNhMDk3NDFlOGJlNDkxOTJmMjdmY2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.XFUgzdVnWE3MY3z3UPGdtQhjEROU-SSIv6OTXFECRF0) ![image](https://private-user-images.githubusercontent.com/63585568/309536298-e544e82b-a4de-4261-985e-ea7bbff8951e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3NDg1OTgsIm5iZiI6MTcwOTc0ODI5OCwicGF0aCI6Ii82MzU4NTU2OC8zMDk1MzYyOTgtZTU0NGU4MmItYTRkZS00MjYxLTk4NWUtZWE3YmJmZjg5NTFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA2VDE4MDQ1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ1MDg0N2JkZWVmN2RlZDIyODQ4OWUwNmEzNWNjN2FiZGIxZmU2MWUxMzA4N2FjZDViZDAyZDI0YjM0MDU0MjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CI0vFInYcQ0X1nVyKYizCKbRYBb4k24u1-Tb5FHvLPk) ![image](https://private-user-images.githubusercontent.com/63585568/309536303-eb77a4cf-9919-4dfc-b52c-d1c897e47960.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3NDg1OTgsIm5iZiI6MTcwOTc0ODI5OCwicGF0aCI6Ii82MzU4NTU2OC8zMDk1MzYzMDMtZWI3N2E0Y2YtOTkxOS00ZGZjLWI1MmMtZDFjODk3ZTQ3OTYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA2VDE4MDQ1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNhMzEzNDYyODdjNDQ0NjAzNDdiYWUyMWM1OTM4NjI4MzFiNTFjNjQ4M2E4NGVlNDk1YmIwZTI1N2IzOGVmYWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.zj84Ri7pxR9XJVUtAPNTjY54b_hnWzrVDX6yXQi8Pqo)

Hi, did you find a solution?

**Comment by phamthaison1207** [comment]
hi there, I finished it and successfully installed mecab, but do I turn this off? when using it?
![image](https://github.com/themoeway/yomitan/assets/63585568/1e4d5f69-9d03-4706-88cb-f16f99484174)

or could I simply just leave this on?

**Comment by avc1657** [comment]
> hi there, I finished it and successfully installed mecab, but do I turn the internal parser off? when using mecab? or could I simply just leave this on? ![image](https://private-user-images.githubusercontent.com/63585568/311422914-1e4d5f69-9d03-4706-88cb-f16f99484174.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5ODE0MjgsIm5iZiI6MTcwOTk4MTEyOCwicGF0aCI6Ii82MzU4NTU2OC8zMTE0MjI5MTQtMWU0ZDVmNjktOWQwMy00NzA2LTg4Y2ItZjE2Zjk5NDg0MTc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA5VDEwNDUyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1ZDQzNTQ4ZjBiYTA3ZmQ3NjE1ZTI1OTMzZjQwNzVhMDYyZDM3N2U1OWEzMjhlNmNiMjc3YWYyMTBiZTUxYzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9c3yAivVIf3y2wmbjTQenD-ztcimNM6iOaTFz93kk9E)

How did you make it work? What did you do differently compared to when you got the error "could not connect to native MeCab content"?

**Comment by peldas** [comment]
@avc1657 I've just resolved the issue myself – involved modifying how the registry key is created in code and also running the Python script under an elevated command prompt. I've raised a PR to the project [here](https://github.com/themoeway/yomitan-mecab-installer/pull/2), but feel free to take a look at the change under the Commits tab and try it yourself locally. I basically gave the line where the registry key is created a few more explicit permissions.
